### PR TITLE
Fix theme redirection

### PR DESCRIPTION
**Pull request by rNiedzwiecki**
There was a problem, that after change theme, application sometimes redirect previous action instead of refreshing. I created a mechanism that should help managing actions to remember.

### CODE REVIEW

File: src/base/BaseController.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+     */
+    public function beforeAction($action): bool
+    {
+        if (!parent::beforeAction($action)) {
+            return false;
+        }
+
+        if (\in_array($action->id, $this->remember(), true)) {
+            $previous = Url::previous('actualUrl');
+            if ($previous !== Url::to('')) {
```
Comment (bizley) [inline src/base/BaseController.php]: Drop `''` as it's default.

File: src/controllers/AdminController.php [hunk 2]
```diff
@@ -48,6 +48,18 @@ public function behaviors(): array
...
 
+    /**
+     * {@inheritdoc}
+     */
+    public function remember(): array
+    {
+        return array_merge(parent::remember(), [
+            'index',
+            'history',
+            'calendar'
```
Comment (bizley) [inline src/controllers/AdminController.php]: Add comma after last element.

File: src/controllers/ClockController.php [hunk 3]
```diff
@@ -49,6 +48,21 @@ public function behaviors(): array
...
+     */
+    public function remember(): array
+    {
+        return array_merge(parent::remember(), [
+            'history',
+            'calendar',
+            'edit',
+            'add',
+            'off-add',
+            'off-edit'
```
Comment (bizley) [inline src/controllers/ClockController.php]: Add comma after last element.

File: src/controllers/ProfileController.php [hunk 4]
```diff
@@ -46,6 +47,18 @@ public function behaviors(): array
...
 
+    /**
+     * {@inheritdoc}
+     */
+    public function remember(): array
+    {
+        return array_merge(parent::remember(), [
+            'index',
+            'api',
+            'pin'
```
Comment (bizley) [inline src/controllers/ProfileController.php]: Add comma after last element.

File: src/controllers/ProfileController.php [hunk 5]
```diff
@@ -174,7 +187,7 @@ public function actionTheme(string $theme): Response
 
         if ($user === null) {
             Yii::$app->alert->danger(Yii::t('app', 'Can not find user of given ID.'));
-        } elseif (!in_array($theme, AppAsset::themes(), true)) {
+        } elseif (!\in_array($theme, AppAsset::themes(), true)) {
```
Comment (bizley) [inline src/controllers/ProfileController.php]: Remove backslash.

File: src/controllers/SiteController.php [hunk 6]
```diff
@@ -60,6 +60,12 @@ public function actions(): array
             'error' => ErrorAction::class,
         ];
     }
+    public function remember(): array
+    {
+        return array_merge(parent::remember(),[
+           'index'
```
Comment (bizley) [inline src/controllers/SiteController.php]: Add comma after last element.

File: src/controllers/SiteController.php [hunk 7]
```diff
@@ -60,6 +60,12 @@ public function actions(): array
             'error' => ErrorAction::class,
         ];
     }
+    public function remember(): array
+    {
+        return array_merge(parent::remember(),[
```
Comment (bizley) [inline src/controllers/SiteController.php]: Add space before `[`.

File: src/base/BaseController.php [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+                Url::remember($previous);
+                Url::remember('', 'actualUrl');
```
Comment (bizley) [inline src/base/BaseController.php]: Change `actualUrl` to `rememberedUrl`.

File: src/base/BaseController.php [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+     * @throws \yii\web\BadRequestHttpException
+     */
+    public function beforeAction($action): bool
+    {
+        if (!parent::beforeAction($action)) {
+            return false;
+        }
+
+        if (\in_array($action->id, $this->remember(), true)) {
+            $previous = Url::previous('actualUrl');
```
Comment (bizley) [inline src/base/BaseController.php]: Change `actualUrl` to `rememberedUrl`.

File: src/controllers/ProfileController.php [hunk 10]
```diff
@@ -140,7 +153,7 @@ public function actionDark(): Response
             }
         }
 
-        return $this->goBack();
+        return $this->redirect(Url::previous('actualUrl'));
```
Comment (bizley) [inline src/controllers/ProfileController.php]: Change `actualUrl` to `rememberedUrl`.

File: src/controllers/ProfileController.php [hunk 11]
```diff
@@ -161,7 +174,7 @@ public function actionLight(): Response
...
```
Comment (bizley) [inline src/controllers/ProfileController.php]: Change `actualUrl` to `rememberedUrl`.

File: src/controllers/ProfileController.php [hunk 12]
```diff
@@ -183,8 +196,7 @@ public function actionTheme(string $theme): Response
                 Yii::$app->alert->danger(Yii::t('app', 'There was an error while saving user.'));
             }
         }
-
-        return $this->goBack();
+        return $this->redirect(Url::previous('actualUrl'));
```
Comment (bizley) [inline src/controllers/ProfileController.php]: Change `actualUrl` to `rememberedUrl`.

**Comment by rNiedzwiecki** [comment]
@bizley changes done

**Pull request merged**